src/service.py

```diff
@@ -0,0 +1,165 @@
+
+    Attrs:
+       is_active: Indicate if the agent service is active and running.
+    """
+
+    def __init__(self, state: State):
+        """Initialize the jenkins agent service.
+
+        Args:
+            state: The Jenkins agent state.
+        """
+        self.state = state
+
+    def _render_file(self, path: str, content: str, mode: int) -> None:
+        """Write a content rendered from a template to a file.
+
+        Args:
+            path: the path to the file.
+            content: the data to be written to the file.
+            mode: access permission mask applied to the
+              file using chmod (e.g. 0o640).
+        """
+        with open(path, "w+", encoding="utf-8") as file:
+            file.write(content)
+        # Ensure correct permissions are set on the file.
+        os.chmod(path, mode)
+        try:
+            # Get the uid/gid for the root user (running the service).
+            # TODO: the user running the jenkins agent is currently root
+            # we should replace this by defining a dedicated user in the apt package
+            u = pwd.getpwnam("root")
+            # Set the correct ownership for the file.
+            os.chown(path, uid=u.pw_uid, gid=u.pw_gid)
+        except KeyError:
+            # Ignore non existing user error when it wasn't created yet.
+            pass
+
+    @property
+    def is_active(self) -> bool:
+        """Indicate if the jenkins agent service is active."""
+        return systemd.service_running(AGENT_SERVICE_NAME)
+
+    def install(self) -> None:
+        """Install and set up the jenkins agent apt package.
+
+        Raises:
+            ServiceRestartError: if the installation of the snap failed.
+            PackageInstallError: if the package installation failed.
+        """
+        try:
+            # Add ppa that hosts the jenkins-agent package
+            repositories = apt.RepositoryMapping()
+            if "deb-ppa.launchpadcontent.net/tphan025/ppa-jammy" not in repositories:
+                repositories.add(
+                    apt.DebianRepository(
+                        enabled=True,
+                        repotype="deb",
+                        uri="https://ppa.launchpadcontent.net/tphan025/ppa/ubuntu/",
+                        # TODO: depends on the series of the charm unit
+                        # set the release accordingly
+                        release="jammy",
+                        groups=["main"],
+                    )
+                )
+                apt.import_key(PPA_GPG_KEY_ID)
+            # Install the apt package
+            apt.update()
+            # Dependencies
+            apt.add_package("openjdk-17-jre")
+            # Jenkins agent package from PPA
+            apt.add_package(AGENT_PACKAGE_NAME)
+            systemd.service_restart(AGENT_SERVICE_NAME)
+        except systemd.SystemdError as exc:
+            raise ServiceRestartError("Error starting the agent service") from exc
```

Comment:
This function is called `install`, should the error messages focus on installing? Based on the function name I wouldn't guess that the install function also starts the service?

Reply:
That makes sense, I removed starting the service from the _on_install hook